### DISCUSSION TITLE
Custom nTicks for each variable in PCP

### DISCUSSION THREAD

**Discussion by Atulsingh92** [anyoptimization/pymoo#241] [Q&A]
Hello, 

I have a following definition of a PCP chart, where all the variables are integer variables, 

```
plot3 = PCP(show_bounds=False, 
            n_ticks=5,
            labels=["ntr","nr","np", "nf", "fincon", "tcon"],
           legend=(True, {'loc': "best"}),
           tight_layout=True,
           figsize=(12,8), fontsize=16) #.add(entry.pop.get("X")).do()
plot3.add(dspace, color="grey", alpha=0.3)
plot3.add(res.X, color="blue", linewidth=3,label="min {}".format(obj_names[0]))#plot3.add(res.X[Imax], color="orange", linewidth=3,label="max {}".format(obj_names[0]))#, obj_names[1]))
plot3.show()
```

Although the 5 ticks work fine if the bound is large, my last two variables are sampled between only 2 and 3 values, so they don't need 5 ticks. 

The question is, how can I customise ticks for each variable, if that is even possible?
(np would need 4, fin con would need 2, and tcon would need 3, rest are fine with 5)

![image](https://user-images.githubusercontent.com/30570817/150144615-9f86e352-9a66-45a2-bc97-2f938abe717f.png)

**Comment by blankjul** [comment]
Yes, you can do that but you will have to write your own `PCP` class at this time. 
So far the PCP code is not too long, so I recommended copying the class and to change this part
```
            if self.n_ticks is not None:
                n_length = 0.03
                for y in np.linspace(0, 1, self.n_ticks):
                    self.ax.hlines(y, i - n_length, i + n_length, **self.axis_style)
```

In your case, `n_ticks` could be a list and with self.n_ticks[i] you could customize the number.
